diff --git a/board/xilinx/zynqmp/zynqmp.c b/board/xilinx/zynqmp/zynqmp.c
index 7a262385bc..6652905f27 100644
--- a/board/xilinx/zynqmp/zynqmp.c
+++ b/board/xilinx/zynqmp/zynqmp.c
@@ -708,3 +708,16 @@ enum env_location env_get_location(enum env_operation op, int prio)
 		return ENVL_NOWHERE;
 	}
 }
+
+#ifdef CONFIG_MV88E61XX_SWITCH
+int mv88e61xx_hw_reset(struct phy_device *phydev)
+{
+	/* Ensure the 88e6185 gets at least 20ms Reset
+	 */
+    gpio_direction_output(54, 1);
+    mdelay(20);
+    gpio_direction_output(54, 0);
+
+	return 0;
+}
+#endif
\ No newline at end of file
diff --git a/drivers/net/phy/Kconfig b/drivers/net/phy/Kconfig
index 51733dd123..c3986855b4 100644
--- a/drivers/net/phy/Kconfig
+++ b/drivers/net/phy/Kconfig
@@ -304,4 +304,10 @@ config PHY_NCSI
 	bool "NC-SI based PHY"
 	depends on DM_ETH
 
+config SWITCH_ETH
+	bool "Switch-Link PHY"
+	depends on DM_ETH
+	help
+	  Marvell MV88E6185 Ethernet switch PHY support.
+
 endif #PHYLIB
diff --git a/drivers/net/phy/Makefile b/drivers/net/phy/Makefile
index 6e722331f1..72e6ea7751 100644
--- a/drivers/net/phy/Makefile
+++ b/drivers/net/phy/Makefile
@@ -33,3 +33,4 @@ obj-$(CONFIG_PHY_VITESSE) += vitesse.o
 obj-$(CONFIG_PHY_MSCC) += mscc.o
 obj-$(CONFIG_PHY_FIXED) += fixed.o
 obj-$(CONFIG_PHY_NCSI) += ncsi.o
+obj-$(CONFIG_SWITCH_ETH) += switch-link.o
diff --git a/drivers/net/phy/mv88e61xx.c b/drivers/net/phy/mv88e61xx.c
index 7eff37b244..9de13e914c 100644
--- a/drivers/net/phy/mv88e61xx.c
+++ b/drivers/net/phy/mv88e61xx.c
@@ -187,6 +187,8 @@
 #define PORT_SWITCH_ID_6240		0x2400
 #define PORT_SWITCH_ID_6250		0x2500
 #define PORT_SWITCH_ID_6352		0x3520
+#define PORT_SWITCH_ID_6185		0x2800
+
 
 struct mv88e61xx_phy_priv {
 	struct mii_dev *mdio_bus;
@@ -420,6 +422,19 @@ static int mv88e61xx_phy_write(struct phy_device *phydev, int phy,
 					    MDIO_DEVAD_NONE, reg, val);
 }
 
+static int mv88e6185_port_read(struct phy_device *phydev, u8 port, u8 reg)
+{
+	struct mv88e61xx_phy_priv *priv = phydev->priv;
+	struct mii_dev *bus;
+	int data;
+	const char *devname;
+    
+	devname = miiphy_get_current_dev();
+	mdelay(10);
+	miiphy_read(devname, priv->port_reg_base + port, reg, &data);
+	return data;
+}
+
 static int mv88e61xx_port_read(struct phy_device *phydev, u8 port, u8 reg)
 {
 	struct mv88e61xx_phy_priv *priv = phydev->priv;
@@ -427,6 +442,18 @@ static int mv88e61xx_port_read(struct phy_device *phydev, u8 port, u8 reg)
 	return mv88e61xx_reg_read(phydev, priv->port_reg_base + port, reg);
 }
 
+static int mv88e6185_port_write(struct phy_device *phydev, u8 port, u8 reg,
+								u16 val)
+{
+	struct mv88e61xx_phy_priv *priv = phydev->priv;
+	struct mii_dev *bus;
+	const char	*devname;
+
+	devname = miiphy_get_current_dev();
+	mdelay(10);
+	return miiphy_write(devname, priv->port_reg_base + port, reg, val);
+}
+
 static int mv88e61xx_port_write(struct phy_device *phydev, u8 port, u8 reg,
 								u16 val)
 {
@@ -758,6 +785,66 @@ static int mv88e61xx_fixed_port_setup(struct phy_device *phydev, u8 port)
 				   val);
 }
 
+static int mv88e6185_fixed_port_setup(struct phy_device *phydev, u8 port)
+{
+	struct mv88e61xx_phy_priv *priv = phydev->priv;
+	int val;
+
+	val = mv88e6185_port_read(phydev, port, PORT_REG_STATUS);
+	val |= SMI_CMD_CLAUSE_22;
+
+	mv88e6185_port_write(phydev, port, PORT_REG_STATUS, val);
+
+	val = mv88e6185_port_read(phydev, port, PORT_REG_PHYS_CTRL);
+	if (val < 0)
+		return val;
+
+	val &= ~(PORT_REG_PHYS_CTRL_SPD_MASK |
+		PORT_REG_PHYS_CTRL_FC_VALUE |
+		PORT_REG_PHYS_CTRL_FC_FORCE);
+
+	val |= PORT_REG_PHYS_CTRL_PCS_AN_EN |
+		PORT_REG_PHYS_CTRL_PCS_AN_RST |
+		PORT_REG_PHYS_CTRL_SPD1000;
+
+	return mv88e6185_port_write(phydev, port, PORT_REG_PHYS_CTRL, val);
+}
+
+static int mv88e6185_set_cpu_port(struct phy_device *phydev)
+{
+	struct mv88e61xx_phy_priv *priv = phydev->priv;
+	int val;
+
+	/* Set CPUDest */
+	val = mv88e61xx_reg_read(phydev, priv->global1, GLOBAL1_MON_CTRL);
+	if (val < 0)
+		return val;
+	val = bitfield_replace(val, GLOBAL1_MON_CTRL_CPUDEST_SHIFT,
+			       GLOBAL1_MON_CTRL_CPUDEST_WIDTH,
+			       CONFIG_MV88E61XX_CPU_PORT);
+	val = mv88e61xx_reg_write(phydev, priv->global1,
+				  GLOBAL1_MON_CTRL, val);
+	if (val < 0)
+		return val;
+
+	/* Enable CPU port */
+	val = mv88e61xx_port_enable(phydev, CONFIG_MV88E61XX_CPU_PORT);
+	if (val < 0)
+		return val;
+
+	val = mv88e61xx_read_port_config(phydev, CONFIG_MV88E61XX_CPU_PORT);
+	if (val < 0)
+		return val;
+
+	val = mv88e6185_fixed_port_setup(phydev,
+					 CONFIG_MV88E61XX_CPU_PORT);
+	if (val < 0)
+		return val;
+
+	return 0;
+}
+
+
 static int mv88e61xx_set_cpu_port(struct phy_device *phydev)
 {
 	struct mv88e61xx_phy_priv *priv = phydev->priv;
@@ -812,6 +899,23 @@ static int mv88e61xx_set_cpu_port(struct phy_device *phydev)
 	return 0;
 }
 
+static int mv88e6185_switch_init(struct phy_device *phydev)
+{
+	static int init;
+	int res;
+
+	if (init)
+		return 0;
+
+	res = mv88e61xx_switch_reset(phydev);
+	if (res < 0)
+		return res;
+
+	init = 1;
+
+	return 0;
+}
+
 static int mv88e61xx_switch_init(struct phy_device *phydev)
 {
 	static int init;
@@ -886,6 +990,36 @@ static int mv88e61xx_phy_config_port(struct phy_device *phydev, u8 phy)
 	return 0;
 }
 
+/*
+ * This function is used to pre-configure the required register
+ * offsets, so that the indirect register access to the PHY registers
+ * is possible. This is necessary to be able to read the PHY ID
+ * while driver probing or in get_phy_id(). The globalN register
+ * offsets must be initialized correctly for a detected switch,
+ * otherwise detection of the PHY ID won't work!
+ */
+static int mv88e6185_priv_reg_offs_pre_init(struct phy_device *phydev)
+{
+	struct mv88e61xx_phy_priv *priv = phydev->priv;
+
+	/*
+	 * Initial 'port_reg_base' value must be an offset of existing
+	 * port register, then reading the ID should succeed. First, try
+	 * to read via port registers with device address 0x10 (88E6096
+	 * and compatible switches).
+	 */
+	priv->port_reg_base = 0x10;
+	priv->id = mv88e61xx_get_switch_id(phydev);
+	if (priv->id != 0xfff0) {
+		priv->global1 = 0x1A;
+		priv->global2 = 0x1B;
+		return 0;
+	}
+
+	debug("%s Unknown ID 0x%x\n", __func__, priv->id);
+	return -ENODEV;
+}
+
 /*
  * This function is used to pre-configure the required register
  * offsets, so that the indirect register access to the PHY registers
@@ -928,6 +1062,91 @@ static int mv88e61xx_priv_reg_offs_pre_init(struct phy_device *phydev)
 	return -ENODEV;
 }
 
+static int mv88e6185_probe(struct phy_device *phydev)
+{
+	struct mii_dev *smi_wrapper;
+	struct mv88e61xx_phy_priv *priv;
+	int res;
+
+	/* TODO: a hardware reset needs to be implemented */
+	/* res = mv88e61xx_hw_reset(phydev);
+	if (res < 0)
+		return res;
+    */
+
+	priv = malloc(sizeof(*priv));
+	if (!priv)
+		return -ENOMEM;
+
+	memset(priv, 0, sizeof(*priv));
+
+	/*
+	 * This device requires indirect reads/writes to the PHY registers
+	 * which the generic PHY code can't handle.  Make a wrapper MII device
+	 * to handle reads/writes
+	 */
+	smi_wrapper = mdio_alloc();
+	if (!smi_wrapper) {
+		free(priv);
+		return -ENOMEM;
+	}
+
+	/*
+	 * Store the mdio bus in the private data, as we are going to replace
+	 * the bus with the wrapper bus
+	 */
+	priv->mdio_bus = phydev->bus;
+
+	/*
+	 * Store the smi bus address in private data.  This lets us use the
+	 * phydev addr field for device address instead, as the genphy code
+	 * expects.
+	 */
+	priv->smi_addr = phydev->addr;
+
+	/*
+	 * Store the phy_device in the wrapper mii device. This lets us get it
+	 * back when genphy functions call phy_read/phy_write.
+	 */
+	smi_wrapper->priv = phydev;
+	strncpy(smi_wrapper->name, "indirect mii", sizeof(smi_wrapper->name));
+	smi_wrapper->read = mv88e61xx_phy_read_indirect;
+	smi_wrapper->write = mv88e61xx_phy_write_indirect;
+
+	/* Replace the bus with the wrapper device */
+	phydev->bus = smi_wrapper;
+
+	phydev->priv = priv;
+
+	res = mv88e6185_priv_reg_offs_pre_init(phydev);
+	if (res < 0)
+		return res;
+
+	debug("%s ID 0x%x\n", __func__, priv->id);
+
+	switch (priv->id) {
+	case PORT_SWITCH_ID_6185:
+		priv->port_count = 10;
+		priv->port_stat_link_mask = BIT(11);
+		priv->port_stat_dup_mask = BIT(8);
+		priv->port_stat_speed_width = 2;
+		priv->phy_ctrl1_en_det_shift = 8;
+		priv->phy_ctrl1_en_det_width = 2;
+		priv->phy_ctrl1_en_det_ctrl =
+			PHY_REG_CTRL1_ENERGY_DET_SENSE_XMIT;
+		break;
+	default:
+		free(priv);
+		return -ENODEV;
+	}
+
+	res = mdio_register(smi_wrapper);
+	if (res)
+		printf("Failed to register SMI bus\n");
+
+	return 0;
+}
+
 static int mv88e61xx_probe(struct phy_device *phydev)
 {
 	struct mii_dev *smi_wrapper;
@@ -1030,6 +1249,36 @@ static int mv88e61xx_probe(struct phy_device *phydev)
 	return 0;
 }
 
+static int mv88e6185_phy_config(struct phy_device *phydev)
+{
+	struct mv88e61xx_phy_priv *priv = phydev->priv;
+	int res;
+	int i;
+	int ret = -1;
+
+	res = mv88e6185_switch_init(phydev);
+	if (res < 0)
+		return res;
+
+	for (i = 0; i < priv->port_count; i++) {
+		if ((1 << i) & CONFIG_MV88E61XX_PHY_PORTS) {
+			phydev->addr = i;
+            /* TODO: functions that require subsequent implementation */
+
+			/* Return success if any PHY succeeds */
+			ret = 0;
+		} else if ((1 << i) & CONFIG_MV88E61XX_FIXED_PORTS) {
+			res = mv88e6185_fixed_port_setup(phydev, i);
+			if (res < 0) {
+				printf("Error configuring port %i\n", i);
+				continue;
+			}
+		}
+	}
+
+	return 0;
+}
+
 static int mv88e61xx_phy_config(struct phy_device *phydev)
 {
 	struct mv88e61xx_phy_priv *priv = phydev->priv;
@@ -1102,6 +1351,12 @@ static int mv88e61xx_phy_is_connected(struct phy_device *phydev)
 	return (val & PHY_REG_STATUS1_ENERGY) == 0;
 }
 
+static int mv88e6185_phy_startup(struct phy_device *phydev)
+{
+    /* TODO: functions that require subsequent implementation */
+	return 0;
+}
+
 static int mv88e61xx_phy_startup(struct phy_device *phydev)
 {
 	struct mv88e61xx_phy_priv *priv = phydev->priv;
@@ -1135,6 +1390,17 @@ static int mv88e61xx_phy_startup(struct phy_device *phydev)
 	return 0;
 }
 
+static struct phy_driver mv88e6185_driver = {
+	.name = "Marvell MV88E6185",
+	.uid = 0x28022803,
+	.mask = 0xfffffff0,
+	.features = PHY_GBIT_FEATURES,
+	.probe = mv88e6185_probe,
+	.config = mv88e6185_phy_config,
+	.startup = mv88e6185_phy_startup,
+	.shutdown = &genphy_shutdown,
+};
+
 static struct phy_driver mv88e61xx_driver = {
 	.name = "Marvell MV88E61xx",
 	.uid = 0x01410eb1,
@@ -1170,6 +1436,7 @@ static struct phy_driver mv88e6071_driver = {
 
 int phy_mv88e61xx_init(void)
 {
+	phy_register(&mv88e6185_driver);
 	phy_register(&mv88e61xx_driver);
 	phy_register(&mv88e609x_driver);
 	phy_register(&mv88e6071_driver);
diff --git a/drivers/net/phy/phy.c b/drivers/net/phy/phy.c
index 9587e6b9fa..9c70666d7a 100644
--- a/drivers/net/phy/phy.c
+++ b/drivers/net/phy/phy.c
@@ -553,6 +553,9 @@ int phy_init(void)
 #endif
 #ifdef CONFIG_PHY_XILINX_GMII2RGMII
 	phy_xilinx_gmii2rgmii_init();
+#endif
+#ifdef CONFIG_SWITCH_ETH
+	phy_switch_init();
 #endif
 	genphy_init();
 
@@ -1002,6 +1005,35 @@ static struct phy_device *phy_connect_fixed(struct mii_dev *bus,
 }
 #endif
 
+#ifdef CONFIG_SWITCH_ETH
+#ifdef CONFIG_DM_ETH
+static struct phy_device *phy_connect_switch(struct mii_dev *bus,
+					    struct udevice *dev,
+					    phy_interface_t interface)
+#else
+static struct phy_device *phy_connect_switch(struct mii_dev *bus,
+					    struct eth_device *dev,
+					    phy_interface_t interface)
+#endif
+{
+	struct phy_device *phydev = NULL;
+	int sn;
+	const char *name;
+
+	sn = fdt_first_subnode(gd->fdt_blob, dev_of_offset(dev));
+	while (sn > 0) {
+		name = fdt_get_name(gd->fdt_blob, sn, NULL);
+		if (name && strcmp(name, "switch-link") == 0) {
+			phydev = phy_device_create(bus, sn, PHY_SWITCH_ID, false,
+						   interface);
+			break;
+		}
+		sn = fdt_next_subnode(gd->fdt_blob, sn);
+	}
+
+	return phydev;
+}
+#endif
 #ifdef CONFIG_DM_ETH
 struct phy_device *phy_connect(struct mii_dev *bus, int addr,
 			       struct udevice *dev,
@@ -1014,14 +1046,12 @@ struct phy_device *phy_connect(struct mii_dev *bus, int addr,
 {
 	struct phy_device *phydev = NULL;
 	uint mask = (addr >= 0) ? (1 << addr) : 0xffffffff;
-
-#ifdef CONFIG_PHY_FIXED
-	phydev = phy_connect_fixed(bus, dev, interface);
+#ifdef CONFIG_SWITCH_ETH
+	phydev = phy_connect_switch(bus, dev, interface);
 #endif
-
-#ifdef CONFIG_PHY_NCSI
+#ifdef CONFIG_PHY_FIXED
 	if (!phydev)
-		phydev = phy_device_create(bus, 0, PHY_NCSI_ID, false, interface);
+			phydev = phy_connect_fixed(bus, dev, interface);
 #endif
 
 #ifdef CONFIG_PHY_XILINX_GMII2RGMII
diff --git a/drivers/net/phy/switch-link.c b/drivers/net/phy/switch-link.c
new file mode 100755
index 0000000000..12031bf5ce
--- /dev/null
+++ b/drivers/net/phy/switch-link.c
@@ -0,0 +1,403 @@
+//SPDX-License-Identifier: GPL-2.0+
+/*
+ * switch-Link phy
+ *
+ * Copyright 2022 zillnk
+ */
+
+#include <config.h>
+#include <common.h>
+#include <phy.h>
+#include <dm.h>
+#include <fdt_support.h>
+#include <asm/gpio.h>
+#include <zynqmp_firmware.h>
+#include <linux/delay.h>
+
+DECLARE_GLOBAL_DATA_PTR;
+
+#define CTC21101							0x782013
+#define TST0101								0x00011a
+#define RTL8211F							0x1cc916
+
+#define CTC21101_CDR_REG_VAL				0x750
+
+#define MIIM_RTL8211F_PAGE_SELECT			0x1f
+#define MIIM_RTL8211F_TX_DELAY				0x100
+
+#define MV88E6185_GLOBAL1_ADDR				0x1b
+#define MV88E6185_PORT_BASE_ADDR			0x10
+#define MV88E6185_NUM_PORTS					0x0a
+#define MV88E6185_PORT_SWITCH_ID			0x03
+#define MV88E6185_GLOBAL1_CTLl				0x04
+
+#define MV88E6185_PORT_SWITCH_ID_PROD_MASK	0xfff0
+#define MV88E6185_PORT_SWITCH_ID_PROD_6185	0x1a70
+
+#define MV88E6185_PORT_STS					0x0000
+#define MV88E6185_PORT_STS_PHY_DETECT		0x1000
+
+#define MV88E6185_PORT_MAC_CTL				0x0001
+#define MV88E6185_PORT_MAC_CTL_LINK_UP		0x0020
+#define MV88E6185_PORT_MAC_CTL_FORCE_LINK	0x0010
+#define MV88E6185_PORT_MAC_CTL_DUPLEX_FULL	0x0008
+#define MV88E6185_PORT_MAC_CTL_FORCE_DUPLEX	0x0004
+#define MV88E6185_PORT_MAC_CTL_SPEED_1000	0x0002
+
+#define MV88E6185_G1_CTL1_PPU_ENABLE		0x4000
+#define MV88E6185_G1_STS_PPU_STATE_MASK		0xc000
+#define MV88E6185_G1_STS_PPU_STATE_DISABLED	0x8000
+#define MV88E6185_G1_STS_PPU_STATE_POLLING	0xc000
+#define MV88E6185_G1_STS_PPU_STATE_BUSY		0x0000
+
+
+static inline int mv88e6185_port_read(struct phy_device *phydev, u8 port, u8 reg)
+{
+	struct mii_dev *bus = phydev->bus;
+	return bus->read(bus, (MV88E6185_PORT_BASE_ADDR + port), MDIO_DEVAD_NONE, reg);
+}
+
+static inline int mv88e6185_port_write(struct phy_device *phydev, u8 port, u8 reg, u16 val)
+{
+	struct mii_dev *bus = phydev->bus;
+	return bus->write(bus, (MV88E6185_PORT_BASE_ADDR + port), MDIO_DEVAD_NONE, reg, val);
+}
+
+static inline void mv88e6xxx_wait_mask(struct phy_device *phydev, u16 mask, u16 val)
+{
+	int i;
+	u16 buf;
+	struct mii_dev *bus = phydev->bus;
+
+	/* There's no bus specific operation to wait for a mask */
+	for (i = 0; i < 16; i++) {
+		buf = bus->read(bus, MV88E6185_GLOBAL1_ADDR, MDIO_DEVAD_NONE,
+		MV88E6185_G1_STS_PPU_STATE_BUSY);
+
+		if ((buf & mask) == val)
+			return;
+
+		mdelay(1);
+	}
+}
+
+static inline void mv88e6185_g1_wait_ppu_disabled(struct phy_device *phydev)
+{
+	mv88e6xxx_wait_mask(phydev, MV88E6185_G1_STS_PPU_STATE_MASK,
+	MV88E6185_G1_STS_PPU_STATE_DISABLED);
+}
+
+static inline void mv88e6185_g1_wait_ppu_polling(struct phy_device *phydev)
+{
+	mv88e6xxx_wait_mask(phydev, MV88E6185_G1_STS_PPU_STATE_MASK,
+	MV88E6185_G1_STS_PPU_STATE_POLLING);
+}
+
+static inline void mv88e6185_g1_ppu_disable(struct phy_device *phydev)
+{
+	u16 val;
+	struct mii_dev *bus = phydev->bus;
+
+	val =  bus->read(bus, MV88E6185_GLOBAL1_ADDR, MDIO_DEVAD_NONE, MV88E6185_GLOBAL1_CTLl);
+	val &= ~MV88E6185_G1_CTL1_PPU_ENABLE;
+	bus->write(bus, MV88E6185_GLOBAL1_ADDR, MDIO_DEVAD_NONE, MV88E6185_GLOBAL1_CTLl, val);
+	mv88e6185_g1_wait_ppu_disabled(phydev);
+}
+
+static inline void mv88e6185_g1_ppu_enable(struct phy_device *phydev)
+{
+	u16 val;
+	struct mii_dev *bus = phydev->bus;
+
+	val =  bus->read(bus, MV88E6185_GLOBAL1_ADDR, MDIO_DEVAD_NONE, MV88E6185_GLOBAL1_CTLl);
+	val |= MV88E6185_G1_CTL1_PPU_ENABLE;
+	bus->write(bus, MV88E6185_GLOBAL1_ADDR, MDIO_DEVAD_NONE, MV88E6185_GLOBAL1_CTLl, val);
+	mv88e6185_g1_wait_ppu_polling(phydev);
+}
+
+static inline int mv88e6185_phy_read(struct phy_device *phydev, u8 port, u8 reg)
+{
+	u16 val;
+	struct mii_dev *bus = phydev->bus;
+
+	mv88e6185_g1_ppu_disable(phydev);
+	val = bus->read(bus, port, MDIO_DEVAD_NONE, reg);
+	mv88e6185_g1_ppu_enable(phydev);
+	return val;
+}
+
+static inline int mv88e6185_phy_write(struct phy_device *phydev, u8 port, u8 reg, u16 val)
+{
+	u16 ret;
+	struct mii_dev *bus = phydev->bus;
+
+	mv88e6185_g1_ppu_disable(phydev);
+	ret = bus->write(bus, port, MDIO_DEVAD_NONE, reg, val);
+	mv88e6185_g1_ppu_enable(phydev);
+	return ret;
+}
+
+static inline int mars_ext_write(struct phy_device *phydev, u8 addr, u32 reg, u16 val)
+{
+	int ret;
+
+	ret = mv88e6185_phy_write(phydev, addr, 0x1e, reg);
+	if (ret < 0)
+		return ret;
+
+	return mv88e6185_phy_write(phydev, addr, 0x1f, val);
+}
+
+static void rtl8211f_phy_setup(struct phy_device *phydev, u8 addr)
+{
+	u16 ctl;
+
+	mv88e6185_phy_write(phydev, addr, MII_BMCR, BMCR_RESET);
+
+	mv88e6185_phy_write(phydev, addr, MIIM_RTL8211F_PAGE_SELECT, 0xd08);
+	ctl = mv88e6185_phy_read(phydev, addr, 0x11);
+	ctl &= ~MIIM_RTL8211F_TX_DELAY;
+
+	mv88e6185_phy_write(phydev, addr, 0x11, ctl);
+	/* restore to default page 0 */
+	mv88e6185_phy_write(phydev, addr, MIIM_RTL8211F_PAGE_SELECT, 0x0);
+
+	/* Set green LED for Link, yellow LED for Active */
+	mv88e6185_phy_write(phydev, addr, MIIM_RTL8211F_PAGE_SELECT, 0xd04);
+	mv88e6185_phy_write(phydev, addr, 0x10, 0x617f);
+	mv88e6185_phy_write(phydev, addr, MIIM_RTL8211F_PAGE_SELECT, 0x0);
+
+	ctl = mv88e6185_port_read(phydev, addr, MV88E6185_PORT_STS);
+	ctl |= MV88E6185_PORT_STS_PHY_DETECT;
+	mv88e6185_port_write(phydev, addr, MV88E6185_PORT_STS, ctl);
+
+	ctl = mv88e6185_phy_read(phydev, addr, MII_CTRL1000);
+	ctl &= ~(ADVERTISE_1000FULL | ADVERTISE_1000HALF);
+	ctl |= ADVERTISE_1000FULL;
+	mv88e6185_phy_write(phydev, addr, MII_CTRL1000, ctl);
+
+	ctl = mv88e6185_phy_read(phydev, addr, MII_BMCR);
+	ctl |= (BMCR_ANENABLE | BMCR_ANRESTART | BMCR_RESET);
+	ctl &= ~(BMCR_ISOLATE);
+
+	mv88e6185_phy_write(phydev, addr, MII_BMCR, ctl);
+}
+
+static void ctc21101_phy_setup(struct phy_device *phydev, u8 addr)
+{
+	u16 ctl;
+
+	mars_ext_write(phydev, addr, 0xa005, 0x0);
+
+	mars_ext_write(phydev, addr, 0xa00d, 0xde00);
+	mars_ext_write(phydev, addr, 0xa00e, 0xd800);
+
+	mars_ext_write(phydev, addr, 0xa000, 0x2);
+	mars_ext_write(phydev, addr, 0x3, CTC21101_CDR_REG_VAL);
+
+	mars_ext_write(phydev, addr, 0xa000, 0x0);
+
+	ctl = mv88e6185_port_read(phydev, addr, MV88E6185_PORT_STS);
+	ctl |= MV88E6185_PORT_STS_PHY_DETECT;
+	mv88e6185_port_write(phydev, addr, MV88E6185_PORT_STS, ctl);
+
+	ctl = mv88e6185_phy_read(phydev, addr, MII_CTRL1000);
+	ctl &= ~(ADVERTISE_1000FULL | ADVERTISE_1000HALF);
+	ctl |= ADVERTISE_1000FULL;
+	mv88e6185_phy_write(phydev, addr, MII_CTRL1000, ctl);
+
+	ctl = mv88e6185_phy_read(phydev, addr, MII_BMCR);
+	ctl |= (BMCR_ANENABLE | BMCR_ANRESTART | BMCR_RESET);
+	ctl &= ~(BMCR_ISOLATE);
+
+	mv88e6185_phy_write(phydev, addr, MII_BMCR, ctl);
+}
+
+static void mii_phy_setup(struct phy_device *phydev, u8 addr)
+{
+	u16 ctl;
+
+	ctl = mv88e6185_port_read(phydev, addr, MV88E6185_PORT_STS);
+	ctl |= MV88E6185_PORT_STS_PHY_DETECT;
+	mv88e6185_port_write(phydev, addr, MV88E6185_PORT_STS, ctl);
+
+	ctl = mv88e6185_phy_read(phydev, addr, MII_CTRL1000);
+	ctl &= ~(ADVERTISE_1000FULL | ADVERTISE_1000HALF);
+	ctl |= ADVERTISE_1000FULL;
+	mv88e6185_phy_write(phydev, addr, MII_CTRL1000, ctl);
+
+	ctl = mv88e6185_phy_read(phydev, addr, MII_BMCR);
+	ctl |= (BMCR_ANENABLE | BMCR_ANRESTART | BMCR_RESET);
+	ctl &= ~(BMCR_ISOLATE);
+	mv88e6185_phy_write(phydev, addr, MII_BMCR, ctl);
+}
+
+static int mv88e6185_phy_port_setup(struct phy_device *phydev, u8 port)
+{
+	u16 val;
+	u32 phy_id;
+
+	val = mv88e6185_phy_read(phydev, port, MII_PHYSID1);
+	phy_id = val;
+	val = mv88e6185_phy_read(phydev, port, MII_PHYSID2);
+	phy_id = (phy_id << 16);
+	phy_id |= val;
+
+	if (phy_id == ~((u32)0U))
+		return -ENOMEM;
+
+	if (phy_id == RTL8211F) {
+		rtl8211f_phy_setup(phydev, port);
+	}
+	else if (phy_id == CTC21101 || phy_id == TST0101) {
+		ctc21101_phy_setup(phydev, port);
+	}
+	else
+		mii_phy_setup(phydev, port);
+
+	return 0;
+}
+
+static void mv88e6185_fixed_port_setup(struct phy_device *phydev, u8 port)
+{
+	u16 val;
+	u16 ctl = 0;
+
+	val = mv88e6185_port_read(phydev, port, MV88E6185_PORT_STS);
+
+	val &= ~(MV88E6185_PORT_STS_PHY_DETECT);
+	mv88e6185_port_write(phydev, port, MV88E6185_PORT_STS, val);
+
+	ctl |= MV88E6185_PORT_MAC_CTL_LINK_UP | MV88E6185_PORT_MAC_CTL_FORCE_LINK |
+		 MV88E6185_PORT_MAC_CTL_DUPLEX_FULL | MV88E6185_PORT_MAC_CTL_FORCE_DUPLEX |
+		 MV88E6185_PORT_MAC_CTL_SPEED_1000;
+	mv88e6185_port_write(phydev, port, MV88E6185_PORT_MAC_CTL, ctl);
+}
+
+static int mv88e6185_hw_reset(struct phy_device *phydev)
+{
+	int ret;
+	struct gpio_desc gpio;
+	u32 reset_gpio;
+	int node = phydev->addr;
+
+	ret = gpio_request_by_name_nodev(offset_to_ofnode(node),
+		  "reset-gpios", 0, &gpio, GPIOD_IS_OUT | GPIOD_IS_OUT_ACTIVE);
+	if (ret) {
+		printf("%s: unable to reset switch device", __func__);
+		return ret;
+	}
+
+	reset_gpio = PMIO_NODE_ID_BASE + gpio.offset;
+
+	/* Request for pin */
+	xilinx_pm_request(PM_PINCTRL_REQUEST, reset_gpio, 0, 0, 0, NULL);
+
+	/* Enable hysteresis in cmos receiver */
+	xilinx_pm_request(PM_PINCTRL_CONFIG_PARAM_SET, reset_gpio,
+			  PM_PINCTRL_CONFIG_SCHMITT_CMOS,
+			  PM_PINCTRL_INPUT_TYPE_SCHMITT, 0, NULL);
+
+	/* Disable Tri-state */
+	xilinx_pm_request(PM_PINCTRL_CONFIG_PARAM_SET, reset_gpio,
+			  PM_PINCTRL_CONFIG_TRI_STATE,
+			  PM_PINCTRL_TRI_STATE_DISABLE, 0, NULL);
+	udelay(1);
+
+	/* Set value 0 to pin */
+	dm_gpio_set_value(&gpio, 0);
+	mdelay(20);
+
+	/* Set value 1 to pin */
+	dm_gpio_set_value(&gpio, 1);
+	mdelay(300);
+	return 0;
+}
+
+int switchphy_probe(struct phy_device *phydev)
+{
+	int i, ret = 0;
+	u32 fixed, phy;
+	struct fixed_link *priv;
+	int ofnode = phydev->addr;
+	unsigned int prod_num;
+
+	priv = malloc(sizeof(*priv));
+	if (!priv)
+		return -ENOMEM;
+	memset(priv, 0, sizeof(*priv));
+
+	phydev->priv = priv;
+	priv->link_speed = SPEED_1000;
+	priv->duplex = DUPLEX_FULL;
+	priv->pause = false;
+	priv->asym_pause = false;
+
+	/* not be reset by core phy code */
+	phydev->flags |= PHY_FLAG_BROKEN_RESET;
+
+	if (mv88e6185_hw_reset(phydev) != 0) {
+		printf("Error on mv88e6185_hw_reset!\n");
+		return -EINVAL;
+	}
+
+	fixed = fdt_getprop_u32_default_node(gd->fdt_blob,
+					   ofnode, 0, "fixed-ports", 0);
+
+	phy = fdt_getprop_u32_default_node(gd->fdt_blob,
+					   ofnode, 0, "phy-ports", 0);
+	if (fixed > 0x1fff || phy > 0x1fff)
+		return -EINVAL;
+
+	ret = mv88e6185_port_read(phydev, 0, MV88E6185_PORT_SWITCH_ID);
+	prod_num = ret & MV88E6185_PORT_SWITCH_ID_PROD_MASK;
+
+	if (prod_num != MV88E6185_PORT_SWITCH_ID_PROD_6185) {
+		printf("Warning: No switch chip!\n");
+		return -EINVAL;
+	}
+
+	for (i = 0; i < MV88E6185_NUM_PORTS; i++) {
+		if ((1 << i) & fixed)
+			mv88e6185_fixed_port_setup(phydev, i);
+
+		if ((1 << i) & phy)
+			mv88e6185_phy_port_setup(phydev, i);
+	}
+
+	return 0;
+}
+
+int switchphy_startup(struct phy_device *phydev)
+{
+	struct fixed_link *priv = phydev->priv;
+
+	phydev->asym_pause = priv->asym_pause;
+	phydev->pause = priv->pause;
+	phydev->duplex = priv->duplex;
+	phydev->speed = priv->link_speed;
+	phydev->link = 1;
+
+	return 0;
+}
+
+int switchphy_shutdown(struct phy_device *phydev)
+{
+	return 0;
+}
+
+static struct phy_driver switchphy_driver = {
+	.uid		= PHY_SWITCH_ID,
+	.mask		= 0xffffffff,
+	.name		= "SWITCH PHY",
+	.features	= PHY_GBIT_FEATURES | SUPPORTED_MII,
+	.probe		= switchphy_probe,
+	.startup	= switchphy_startup,
+	.shutdown	= switchphy_shutdown,
+};
+
+int phy_switch_init(void)
+{
+	phy_register(&switchphy_driver);
+	return 0;
+}
diff --git a/drivers/net/zynq_gem.c b/drivers/net/zynq_gem.c
index 2e92cb5727..f921502612 100644
--- a/drivers/net/zynq_gem.c
+++ b/drivers/net/zynq_gem.c
@@ -480,7 +480,8 @@ static int zynq_gem_init(struct udevice *dev)
 		 * Must be written after PCS_SEL is set in nwconfig,
 		 * otherwise writes will not take effect.
 		 */
-		if (priv->phydev->phy_id != PHY_FIXED_ID)
+		if (priv->phydev->phy_id != PHY_FIXED_ID &&
+			priv->phydev->phy_id != PHY_SWITCH_ID)
 			writel(readl(&regs->pcscntrl) | ZYNQ_GEM_PCS_CTL_ANEG_ENBL,
 			       &regs->pcscntrl);
 		else
diff --git a/include/phy.h b/include/phy.h
index cbdb10d6fc..850281bd06 100644
--- a/include/phy.h
+++ b/include/phy.h
@@ -23,6 +23,7 @@ struct udevice;
 
 #define PHY_FIXED_ID		0xa5a55a5a
 #define PHY_NCSI_ID            0xbeefcafe
+#define PHY_SWITCH_ID		0x5a5aa5a5
 
 /*
  * There is no actual id for this.
@@ -511,6 +512,7 @@ int phy_mscc_init(void);
 int phy_fixed_init(void);
 int phy_ncsi_init(void);
 int phy_xilinx_gmii2rgmii_init(void);
+int phy_switch_init(void);
 
 int board_phy_config(struct phy_device *phydev);
 int get_phy_id(struct mii_dev *bus, int addr, int devad, u32 *phy_id);
