From ba6fe0b75827867e306ccc8e0a3517dda33c0adc Mon Sep 17 00:00:00 2001
From: sbjf0005 <chao.fang@zillnk.com>
Date: Sun, 17 Jul 2022 17:54:27 +0800
Subject: [PATCH] [FDD] support 3types GigaDevice norflash in uboot

---
 drivers/mtd/spi/spi-nor-ids.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/mtd/spi/spi-nor-ids.c b/drivers/mtd/spi/spi-nor-ids.c
index 25d07d01..40f01ae2 100644
--- a/drivers/mtd/spi/spi-nor-ids.c
+++ b/drivers/mtd/spi/spi-nor-ids.c
@@ -122,6 +122,21 @@ const struct flash_info spi_nor_ids[] = {
 		INFO("gd25lx256e", 0xc86819, 0, 64 * 1024, 512,
 		     SECT_4K | SPI_NOR_OCTAL_READ | SPI_NOR_4B_OPCODES)
 	},
+        {
+                INFO("gd55b02ge", 0xc8471c, 0, 64 * 1024, 4096,
+                     SECT_4K | SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES |
+                         SPI_NOR_HAS_LOCK | SPI_NOR_DUAL_READ | SPI_NOR_HAS_TB)
+        },
+        {
+                INFO("gd25b256me", 0xc8471a, 0, 64 * 1024, 1024,
+                     SECT_4K | SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES |
+                         SPI_NOR_HAS_LOCK | SPI_NOR_DUAL_READ | SPI_NOR_HAS_TB)
+        },
+        {
+                INFO("gd55b01ge", 0xc8471b, 0, 64 * 1024, 2048,
+                     SECT_4K | SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES |
+                         SPI_NOR_HAS_LOCK | SPI_NOR_DUAL_READ | SPI_NOR_HAS_TB)
+        },
 #endif
 #ifdef CONFIG_SPI_FLASH_ISSI		/* ISSI */
 	/* ISSI */
-- 
2.31.1

