diff --git a/board/xilinx/zynqmp/zynqmp.c b/board/xilinx/zynqmp/zynqmp.c
index 9930d11a23..e2d3c59f45 100644
--- a/board/xilinx/zynqmp/zynqmp.c
+++ b/board/xilinx/zynqmp/zynqmp.c
@@ -57,6 +57,9 @@ enum {
 	ZYNQMP_VARIANT_EV = BIT(1U),
 	ZYNQMP_VARIANT_CG = BIT(2U),
 	ZYNQMP_VARIANT_DR = BIT(3U),
+	ZYNQMP_VARIANT_DR_SE = BIT(4),
+	ZYNQMP_VARIANT_EG_SE = BIT(5),
+	ZYNQMP_VARIANT_TEG = BIT(6),	
 };
 
 static const struct {
@@ -122,6 +125,11 @@ static const struct {
 		.device = 19,
 		.variants = ZYNQMP_VARIANT_EG,
 	},
+	{
+		.id = 0x0475C093,
+		.device = 19,
+		.variants = ZYNQMP_VARIANT_EG_SE,
+	},	
 	{
 		.id = 0x047E1093,
 		.device = 21,
@@ -172,6 +180,11 @@ static const struct {
 		.device = 47,
 		.variants = ZYNQMP_VARIANT_DR,
 	},
+	{
+		.id = 0x047FA093,
+		.device = 47,
+		.variants = ZYNQMP_VARIANT_DR_SE,
+	},	
 	{
 		.id = 0x047FB093,
 		.device = 48,
@@ -187,6 +200,11 @@ static const struct {
 		.device = 67,
 		.variants = ZYNQMP_VARIANT_DR,
 	},
+	{
+		.id = 0x046d7093,
+		.device = 67,
+		.variants = ZYNQMP_VARIANT_DR_SE,
+	},	
 };
 
 static const struct {
@@ -300,6 +318,10 @@ static char *zynqmp_get_silicon_idcode_name(void)
 		strncat(name, "eg", 2);
 	} else if (zynqmp_devices[i].variants & ZYNQMP_VARIANT_DR) {
 		strncat(name, "dr", 2);
+	} else if (zynqmp_devices[i].variants & ZYNQMP_VARIANT_EG_SE) {
+		strncat(name, "eg_SE", 5);		
+	} else if (zynqmp_devices[i].variants & ZYNQMP_VARIANT_DR_SE) {
+		strncat(name, "dr_SE", 5);
 	} else {
 		debug("Variant not identified\n");
 	}
