diff --git a/drivers/fpga/xilinx.c b/drivers/fpga/xilinx.c
index cbebefb5..380a726c 100644
--- a/drivers/fpga/xilinx.c
+++ b/drivers/fpga/xilinx.c
@@ -81,7 +81,7 @@ int fpga_loadbitstream(int devnum, char *fpgadata, size_t size,
 
 	length = (*dataptr << 8) + *(dataptr + 1);
 	dataptr += 2;
-
+#if 0
 	if (xdesc->name) {
 		i = (ulong)strstr((char *)dataptr, xdesc->name);
 		if (!i) {
@@ -95,6 +95,7 @@ int fpga_loadbitstream(int devnum, char *fpgadata, size_t size,
 		printf("%s: Please fill correct device ID to xilinx_desc\n",
 		       __func__);
 	}
+#endif
 	printf("  part number = \"%s\"\n", dataptr);
 	dataptr += length;
 
