diff --git a/common/bootm.c b/common/bootm.c
index 167eea4a1e..00c4dc4426 100644
--- a/common/bootm.c
+++ b/common/bootm.c
@@ -397,15 +397,16 @@ static int bootm_load_os(bootm_headers_t *images, int boot_progress)
 
 	flush_cache(flush_start, ALIGN(load_end, ARCH_DMA_MINALIGN) - flush_start);
 
-	debug("   kernel loaded at 0x%08lx, end = 0x%08lx\n", load, load_end);
+	printf("------wufan---------   kernel loaded at 0x%08lx, end = 0x%08lx\n", load, load_end);
+	printf("------wufan---------   kernel loaded from 0x%08lx, len = 0x%08lx\n", image_buf, image_len);
 	bootstage_mark(BOOTSTAGE_ID_KERNEL_LOADED);
 
 	no_overlap = (os.comp == IH_COMP_NONE && load == image_start);
 
 	if (!no_overlap && load < blob_end && load_end > blob_start) {
-		debug("images.os.start = 0x%lX, images.os.end = 0x%lx\n",
+		printf("-----wufan-----images.os.start = 0x%lX, images.os.end = 0x%lx\n",
 		      blob_start, blob_end);
-		debug("images.os.load = 0x%lx, load_end = 0x%lx\n", load,
+		printf("------wufan----images.os.load = 0x%lx, load_end = 0x%lx\n", load,
 		      load_end);
 
 		/* Check what type of image this is. */
