diff --git a/bl31/bl31_main.c b/bl31/bl31_main.c
index 44bf32cb7..6de9ee68c 100644
--- a/bl31/bl31_main.c
+++ b/bl31/bl31_main.c
@@ -162,6 +162,7 @@ void bl31_main(void)
 	 * from BL31
 	 */
 	bl31_plat_runtime_setup();
+	NOTICE("BL31: bl31_plat_runtime_setup called.\n");
 }
 
 /*******************************************************************************
@@ -212,7 +213,7 @@ void __init bl31_prepare_next_image_entry(void)
 	assert(next_image_info != NULL);
 	assert(image_type == GET_SECURITY_STATE(next_image_info->h.attr));
 
-	INFO("BL31: Preparing for EL3 exit to %s world\n",
+	NOTICE("BL31: Preparing for EL3 exit to %s world\n",
 		(image_type == SECURE) ? "secure" : "normal");
 	print_entry_point_info(next_image_info);
 	cm_init_my_context(next_image_info);
