diff --git a/include/configs/xilinx_versal.h b/include/configs/xilinx_versal.h
index 8caf5394ed..5abd21df50 100644
--- a/include/configs/xilinx_versal.h
+++ b/include/configs/xilinx_versal.h
@@ -10,6 +10,7 @@
 #ifndef __XILINX_VERSAL_H
 #define __XILINX_VERSAL_H
 
+#include <zboot_info.h>
 /* Generic Interrupt Controller Definitions */
 #define GICD_BASE	0xF9000000
 #define GICR_BASE	0xF9080000
@@ -34,6 +35,8 @@
 #endif
 
 #define ENV_MEM_LAYOUT_SETTINGS \
+	"initrd_high=0x28000000\0" \
+	"zboot=1\0" \
 	"fdt_addr_r=0x40000000\0" \
 	"fdt_size_r=0x400000\0" \
 	"pxefile_addr_r=0x10000000\0" \
@@ -41,9 +44,13 @@
 	"kernel_size_r=0x10000000\0" \
 	"kernel_comp_addr_r=0x30000000\0" \
 	"kernel_comp_size=0x3C00000\0" \
-	"scriptaddr=0x20000000\0" \
+	"scriptaddr=0x2000000\0" \
 	"ramdisk_addr_r=0x02100000\0" \
-	"script_size_f=0x80000\0"
+	"script_offset_f=0x7e0000\0" \
+	"script_size_f=0x20000\0" \
+	"zboot_info=" ZBOOT_INFO "\0" \
+	"zboot_ver=" ZBOOT_VERSION "\0" \
+	"bypass_fpga_load=yes\0"
 
 #if defined(CONFIG_MMC_SDHCI_ZYNQ)
 # define BOOT_TARGET_DEVICES_MMC(func)	func(MMC, mmc, 0) func(MMC, mmc, 1)
