diff --git a/boot/bootm.c b/boot/bootm.c
index a4c0870c0f..64e85c8acc 100644
--- a/boot/bootm.c
+++ b/boot/bootm.c
@@ -427,15 +427,16 @@ static int bootm_load_os(struct bootm_headers *images, int boot_progress)
 
 	flush_cache(flush_start, ALIGN(load_end, ARCH_DMA_MINALIGN) - flush_start);
 
-	debug("   kernel loaded at 0x%08lx, end = 0x%08lx\n", load, load_end);
+	printf("------Zillnk_DEBUG---------   kernel loaded at 0x%08lx, end = 0x%08lx\n", load, load_end);
+	printf("------Zillnk_DEBUG---------   kernel loaded from 0x%08lx, len = 0x%08lx\n", image_buf, image_len);
 	bootstage_mark(BOOTSTAGE_ID_KERNEL_LOADED);
 
 	no_overlap = (os.comp == IH_COMP_NONE && load == image_start);
 
 	if (!no_overlap && load < blob_end && load_end > blob_start) {
-		debug("images.os.start = 0x%lX, images.os.end = 0x%lx\n",
+		printf("-----Zillnk_DEBUG-----images.os.start = 0x%lX, images.os.end = 0x%lx\n",
 		      blob_start, blob_end);
-		debug("images.os.load = 0x%lx, load_end = 0x%lx\n", load,
+		printf("------Zillnk_DEBUG----images.os.load = 0x%lx, load_end = 0x%lx\n", load,
 		      load_end);
 
 		/* Check what type of image this is. */
