From 979ff25afe99ff2d401fb10affa2e06d7c8901fa Mon Sep 17 00:00:00 2001
From: Fan Wu <fan.wu@zillnk.com>
Date: Mon, 3 Nov 2025 03:37:46 +0000
Subject: [PATCH] test ptp

---
 drivers/ptp/ptp_xilinx.c | 22 ++++++++++++++--------
 1 file changed, 14 insertions(+), 8 deletions(-)

diff --git a/drivers/ptp/ptp_xilinx.c b/drivers/ptp/ptp_xilinx.c
index ce3ae8684341..77ff59aa03d9 100644
--- a/drivers/ptp/ptp_xilinx.c
+++ b/drivers/ptp/ptp_xilinx.c
@@ -78,12 +78,15 @@
 /* I/O accessors */
 static inline u32 xlnx_ptp_ior(struct xlnx_ptp_timer *timer, off_t reg)
 {
-	return ioread32(timer->baseaddr + reg);
+	u32 value = ioread32(timer->baseaddr + reg);
+	printk("-------wufan------ %s reg[0x%x] value[0x%x]\n", __func__, (timer->baseaddr + reg), value);
+	return value;
 }
 
 static inline void xlnx_ptp_iow(struct xlnx_ptp_timer *timer, off_t reg,
 				u32 value)
 {
+	printk("-------wufan------ %s reg[0x%x] value[0x%x]\n", __func__, (timer->baseaddr + reg), value);
 	iowrite32(value, (timer->baseaddr + reg));
 }
 
@@ -391,7 +394,7 @@ static int xlnx_ptp_timer_probe(struct platform_device *pdev)
 	int err = 0;
 	struct timespec64 ts, tsp;
 	u32 nsec, sech, secl;
-
+	printk("-------------wufan----------- %s Enter.\n", __func__);
 	if (of_device_is_compatible(pdev->dev.of_node, "xlnx,timer-syncer-1588-1.0")) {
 		dev_err(&pdev->dev, "This Beta version is no longer supported. Please upgrade to 2.0 IP version and use xlnx,timer-syncer-1588-2.0 compatible string\n");
 		return -EINVAL;
@@ -430,7 +433,7 @@ static int xlnx_ptp_timer_probe(struct platform_device *pdev)
 	spin_lock_init(&timer->reg_lock);
 
 	timer->ptp_clock_info = xlnx_ptp_clock_info;
-
+	xlnx_ptp_ior(timer, XPTPTIMER_SYS_NS_OFFSET);printk("-------------wufan----------- %s ptp_clock_register Enter.\n", __func__);
 	timer->ptp_clock = ptp_clock_register(&timer->ptp_clock_info,
 					      &pdev->dev);
 	if (IS_ERR(timer->ptp_clock)) {
@@ -438,12 +441,12 @@ static int xlnx_ptp_timer_probe(struct platform_device *pdev)
 		dev_err(&pdev->dev, "Failed to register ptp clock\n");
 		return err;
 	}
-
+	printk("-------------wufan----------- %s ptp_clock_register Next.\n", __func__);
 	if (timer->irq > 0)
 		/* Enable timer interrupt */
 		xlnx_ptp_iow(timer, XPTPTIMER_IER_OFFSET,
 			     XPTPTIMER_EXTS_1PPS_INTR_MASK);
-
+	printk("-------------wufan----------- %s of_device_is_compatible Enter.\n", __func__);
 	timer->period_0 = TOD_LEGACY_SYS_PERIOD_0;
 	timer->period_1 = TOD_LEGACY_SYS_PERIOD_1;
 
@@ -451,7 +454,7 @@ static int xlnx_ptp_timer_probe(struct platform_device *pdev)
 		timer->period_0 = TOD_SYS_PERIOD_0;
 		timer->period_1 = TOD_SYS_PERIOD_1;
 	}
-
+    printk("-------------wufan----------- %s xlnx_ptp_iow Enter.\n", __func__);
 	xlnx_ptp_iow(timer, XPTPTIMER_TOD_CONFIG_OFFSET,
 		     XPTPTIMER_CFG_MAIN_TOD_EN | XPTPTIMER_CFG_ENABLE_PORT0);
 	/*
@@ -461,9 +464,10 @@ static int xlnx_ptp_timer_probe(struct platform_device *pdev)
 	 * so that the initial LOAD triggers everything together.
 	 */
 	timer->incr = ((u64)XPTPTIMER_CLOCK_PERIOD << XPTPTIMER_PERIOD_SHIFT);
+	printk("-------------wufan----------- %s xlnx_tod_period_write Enter.\n", __func__);
 	xlnx_tod_period_write(timer, timer->incr);
 	xlnx_port_period_write(timer, timer->incr);
-
+	printk("-------------wufan----------- %s xlnx_ptp_settime Enter.\n", __func__);
 	/* Initialize current time */
 	ts = ns_to_timespec64(ktime_to_ns(ktime_get_real()));
 	xlnx_ptp_settime(&timer->ptp_clock_info, &ts);
@@ -472,6 +476,7 @@ static int xlnx_ptp_timer_probe(struct platform_device *pdev)
 	 * A static delay of 7-8 clock cycles is expected between system and
 	 * port timer which should be 28-32ns on this system with 250MHz clock.
 	 */
+	printk("-------------wufan----------- %s xlnx_ptp_iow Enter.\n", __func__);
 	xlnx_ptp_iow(timer, XPTPTIMER_TOD_SNAPSHOT_OFFSET,
 		     XPTPTIMER_SNAPSHOT_MASK);
 	nsec = xlnx_ptp_ior(timer, XPTPTIMER_SYS_NS_OFFSET);
@@ -489,10 +494,11 @@ static int xlnx_ptp_timer_probe(struct platform_device *pdev)
 	ts = timespec64_sub(ts, tsp);
 	timer->static_delay = timespec64_to_ns(&ts);
 	dev_dbg(&pdev->dev, "Static delay %d\n", timer->static_delay);
-
+	printk("-------------wufan----------- %s platform_set_drvdata Enter.\n", __func__);
 	platform_set_drvdata(pdev, timer);
 
 	timer->phc_index = ptp_clock_index(timer->ptp_clock);
+	printk("-------------wufan----------- %s  Exit.\n", __func__);
 	dev_info(&pdev->dev, "Xilinx PTP timer driver probed\n");
 
 	return 0;
-- 
2.34.1

