diff --git a/lib/sw_services/xilplmi/src/versal/xplmi_err.c b/lib/sw_services/xilplmi/src/versal/xplmi_err.c
index cdcc6ac412..d4ad64ef6f 100644
--- a/lib/sw_services/xilplmi/src/versal/xplmi_err.c
+++ b/lib/sw_services/xilplmi/src/versal/xplmi_err.c
@@ -318,8 +318,7 @@ static XPlmi_Error_t ErrorTable[XPLMI_ERROR_SW_ERR_MAX] = {
 	{ .Handler = XPlmi_ErrPrintToLog,
 			.Action = XPLMI_EM_ACTION_PRINT_TO_LOG, .SubsystemId = 0U, },
 	[XPLMI_ERROR_FPD_SWDT] =
-	{ .Handler = XPlmi_ErrPrintToLog,
-			.Action = XPLMI_EM_ACTION_PRINT_TO_LOG, .SubsystemId = 0U, },
+	{ .Handler = NULL, .Action = XPLMI_EM_ACTION_SRST, .SubsystemId = 0U, },
 	[XPLMI_ERROR_PSM_RSRV4] =
 	{ .Handler = NULL, .Action = XPLMI_EM_ACTION_INVALID, .SubsystemId = 0U, },
 	[XPLMI_ERROR_PSM_RSRV5] =
