diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c
index df3449719d..91b107fb63 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c
@@ -121,7 +121,10 @@ int main(void )
 				/* Get Start time for Boot Device init. */
 				XTime_GetTime(&tCur);
 #endif
-
+				XFsbl_Printf(DEBUG_INFO,
+						"FSBL system error! Initialization failed 0x%0lx\n\r", XFSBL_STAGE_ERR);
+				FsblStatus = XFSBL_STAGE_ERR;
+				return -1;
 				XFsbl_Printf(DEBUG_INFO,
 						"================= In Stage 2 ============ \n\r");
 
