diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_handoff.c b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_handoff.c
index f0db0692b6..b8aac3f01a 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_handoff.c
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_handoff.c
@@ -553,6 +553,8 @@ void XFsbl_HandoffExit(u64 HandoffAddress, u32 Flags)
 	RegVal |= XFSBL_EXEC_COMPLETED;
 	XFsbl_Out32(PMU_GLOBAL_GLOB_GEN_STORAGE5, RegVal);
 
+	XFsbl_Printf(DEBUG_GENERAL,"FPGA PL reg 0x1006[0x%08x], 0x1007[0x%08x]. \n\r", XFsbl_In32(0xa0004018), XFsbl_In32(0xa000401c));
+	XFsbl_Printf(DEBUG_GENERAL,"HandoffAddress: 0x%016x, Flags: %d.\n\r", HandoffAddress, Flags);
 	XFsbl_Printf(DEBUG_HANDOFF,"Exit from FSBL \n\r");
 
 	/**
