diff --git a/board/xilinx/zynqmp/zynqmp.c b/board/xilinx/zynqmp/zynqmp.c
index d07979f0dd3..88a1b924d34 100644
--- a/board/xilinx/zynqmp/zynqmp.c
+++ b/board/xilinx/zynqmp/zynqmp.c
@@ -354,6 +354,9 @@ static int reset_reason(void)
 	puts("\n");
 
 	env_set("reset_reason", reason);
+	ret = zynqmp_mmio_write((ulong)&crlapb_base->reset_reason, ~0, ~0);
+	if (ret)
+		return -EINVAL;
 
 	return 0;
 }
@@ -401,10 +404,6 @@ static int boot_targets_setup(void)
 	struct udevice *dev;
 	int bootseq = -1;
 	int bootseq_len = 0;
-	int env_targets_len = 0;
-	const char *mode = NULL;
-	char *new_targets;
-	char *env_targets;
 
 	bootmode = zynqmp_get_bootmode();
 
@@ -412,17 +411,14 @@ static int boot_targets_setup(void)
 	switch (bootmode) {
 	case USB_MODE:
 		puts("USB_MODE\n");
-		mode = "usb_dfu0 usb_dfu1";
 		env_set("modeboot", "usb_dfu_spl");
 		break;
 	case JTAG_MODE:
 		puts("JTAG_MODE\n");
-		mode = "jtag pxe dhcp";
 		env_set("modeboot", "jtagboot");
 		break;
 	case QSPI_MODE_24BIT:
 	case QSPI_MODE_32BIT:
-		mode = "qspi0";
 		puts("QSPI_MODE\n");
 		env_set("modeboot", "qspiboot");
 		break;
@@ -437,7 +433,6 @@ static int boot_targets_setup(void)
 		}
 		debug("mmc0 device found at %p, seq %d\n", dev, dev_seq(dev));
 
-		mode = "mmc";
 		bootseq = dev_seq(dev);
 		env_set("modeboot", "emmcboot");
 		break;
@@ -452,7 +447,6 @@ static int boot_targets_setup(void)
 		}
 		debug("mmc0 device found at %p, seq %d\n", dev, dev_seq(dev));
 
-		mode = "mmc";
 		bootseq = dev_seq(dev);
 		env_set("modeboot", "sdboot");
 		break;
@@ -470,13 +464,11 @@ static int boot_targets_setup(void)
 		}
 		debug("mmc1 device found at %p, seq %d\n", dev, dev_seq(dev));
 
-		mode = "mmc";
 		bootseq = dev_seq(dev);
 		env_set("modeboot", "sdboot");
 		break;
 	case NAND_MODE:
 		puts("NAND_MODE\n");
-		mode = "nand0";
 		env_set("modeboot", "nandboot");
 		break;
 	default:
@@ -484,35 +476,10 @@ static int boot_targets_setup(void)
 		break;
 	}
 
-	if (mode) {
-		if (bootseq >= 0) {
-			bootseq_len = snprintf(NULL, 0, "%i", bootseq);
-			debug("Bootseq len: %x\n", bootseq_len);
-			env_set_hex("bootseq", bootseq);
-		}
-
-		/*
-		 * One terminating char + one byte for space between mode
-		 * and default boot_targets
-		 */
-		env_targets = env_get("boot_targets");
-		if (env_targets)
-			env_targets_len = strlen(env_targets);
-
-		new_targets = calloc(1, strlen(mode) + env_targets_len + 2 +
-				     bootseq_len);
-		if (!new_targets)
-			return -ENOMEM;
-
-		if (bootseq >= 0)
-			sprintf(new_targets, "%s%x %s", mode, bootseq,
-				env_targets ? env_targets : "");
-		else
-			sprintf(new_targets, "%s %s", mode,
-				env_targets ? env_targets : "");
-
-		env_set("boot_targets", new_targets);
-		free(new_targets);
+	if (bootseq >= 0) {
+		bootseq_len = snprintf(NULL, 0, "%i", bootseq);
+		debug("Bootseq len: %x\n", bootseq_len);
+		env_set_hex("bootseq", bootseq);
 	}
 
 	return 0;
@@ -530,14 +497,6 @@ int board_late_init(void)
 	if (multiboot >= 0)
 		env_set_hex("multiboot", multiboot);
 
-	if (!(gd->flags & GD_FLG_ENV_DEFAULT)) {
-		debug("Saved variables - Skipping\n");
-		return 0;
-	}
-
-	if (!IS_ENABLED(CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG))
-		return 0;
-
 	ret = set_fdtfile();
 	if (ret)
 		return ret;
diff --git a/include/configs/xilinx_zynqmp.h b/include/configs/xilinx_zynqmp.h
index 13941ba3589..f7b9d890368 100644
--- a/include/configs/xilinx_zynqmp.h
+++ b/include/configs/xilinx_zynqmp.h
@@ -45,6 +45,8 @@
 /* Ethernet driver */
 
 #define ENV_MEM_LAYOUT_SETTINGS \
+	"initrd_high=0x38000000\0" \
+	"zboot=1\0" \
 	"fdt_addr_r=0x40000000\0" \
 	"fdt_size_r=0x400000\0" \
 	"pxefile_addr_r=0x10000000\0" \
