diff --git a/cmd/zboot.c b/cmd/zboot.c
index ceb1b3c8b03..d42b9ca41ed 100644
--- a/cmd/zboot.c
+++ b/cmd/zboot.c
@@ -174,6 +174,15 @@ static long get_fpga_device(char *arg)
 	return dev;
 }
 
+static int process_bypass_fpga_load(void)
+{
+    const char *val = env_get("bypass_fpga_load");
+    if (val && strcmp(val, "yes") == 0) {
+        return 1;
+    }
+	return 0;
+}
+
 static int do_fpga_loads_rsa_bitstream(int devnum, uint32_t fpga_data, uint32_t data_size)
 {
 	struct fpga_secure_info fpga_sec_info;
@@ -306,7 +315,7 @@ static int parse_head_data(uint32_t base_addr, uint8_t fpganum)
 	//printf("%s: bit2_offset : %#x\t bit2_size: %#x\n", __func__, head_data.bit2_offset, head_data.bit2_size);
 	for (uint8_t i = 0; i < fpganum; i++) {
 		printf("%s: bit[%d]_offset : %#x\t bit[%d]_size: %#x\n", __func__, 2+i, head_data.lastmbh[i].mbit_off, 2+i, head_data.lastmbh[i].mbit_sz);
-	}	
+	}
 #endif
 
 }
@@ -448,6 +457,11 @@ static int do_zboot(struct cmd_tbl *cmdtp, int flag, int argc, char * const argv
 
 	sh = (slot_head *)dataptr;
 
+	if(process_bypass_fpga_load()) {
+		printf("bypass_fpga_load is set to 'yes', proceeding with FPGA loading bypass.\n");
+		goto kernel_loading;
+	}
+
 #ifdef CONFIG_ZU67_SELECTION
 	int zu67_selection = 1;//set standard zu67 as default.
 	int zu67_idx = 0;
@@ -460,9 +474,9 @@ static int do_zboot(struct cmd_tbl *cmdtp, int flag, int argc, char * const argv
 	}else {
 		zu67_idx = 1;
 		printf("Limited zu67, to select fpga[%d]\n", zu67_idx);
-		if (fpganum == 1 
+		if (fpganum == 1
 			|| fpganum == 0
-			|| sh->mbh[zu67_idx - 1].mbit_off == 0xFFFFFFFF 
+			|| sh->mbh[zu67_idx - 1].mbit_off == 0xFFFFFFFF
 			|| sh->mbh[zu67_idx - 1].mbit_sz == 0xFFFFFFFF
 			|| sh->mbh[zu67_idx - 1].mbit_off == 0x0
 			|| sh->mbh[zu67_idx - 1].mbit_sz == 0x0) {
@@ -503,9 +517,9 @@ static int do_zboot(struct cmd_tbl *cmdtp, int flag, int argc, char * const argv
 	}
 	printf("End HDID bin selection %#x\n", bin_selection, gpio_get_value(10));
     /*
-			HDID		BIT3	BIT2	BIT1	BIT0	
+			HDID		BIT3	BIT2	BIT1	BIT0
 			MIO			MIO10	MIO9	MIO8	MIO7	Hex
-		"Macro1(NarrowBand PA Solution)"	
+		"Macro1(NarrowBand PA Solution)"
 			N25N66		0	0	0	0	0x0
 			N7N30		0	0	0	1	0x1
 			N26N29		0	0	1	0	0x2
@@ -514,7 +528,7 @@ static int do_zboot(struct cmd_tbl *cmdtp, int flag, int argc, char * const argv
 			TBD			0	1	0	1	0x5
 			TBD			0	1	1	0	0x6
 			TBD			0	1	1	1	0x7
-		"Macro2(BroadBand PA Solutino)"	
+		"Macro2(BroadBand PA Solutino)"
 			N26N29		1	0	0	0	0x8
 			N8N20N28B	1	0	0	1	0x9
 			B12B13N71	1	0	1	0	0xA
@@ -527,15 +541,15 @@ static int do_zboot(struct cmd_tbl *cmdtp, int flag, int argc, char * const argv
 
 	fpga_idx = (bin_selection >> 3) & 0x1;
 	printf("fpga index [%d] from GPIO <--> binary header fpga num [%d]\n", fpga_idx, fpganum);
-	if ( fpga_idx >= 1 ) 
+	if ( fpga_idx >= 1 )
 	{
-		if (fpganum == 1 
+		if (fpganum == 1
 			|| fpganum == 0
-			|| sh->mbh[fpga_idx - 1].mbit_off == 0xFFFFFFFF 
+			|| sh->mbh[fpga_idx - 1].mbit_off == 0xFFFFFFFF
 			|| sh->mbh[fpga_idx - 1].mbit_sz == 0xFFFFFFFF
 			|| sh->mbh[fpga_idx - 1].mbit_off == 0x0
 			|| sh->mbh[fpga_idx - 1].mbit_sz == 0x0) {
-			
+
 			if (fpganum == 1 || fpganum == 0){
 				printf("Warn: old style package with ONLY fpga0!\n");
 			} else {
@@ -558,7 +572,7 @@ static int do_zboot(struct cmd_tbl *cmdtp, int flag, int argc, char * const argv
 	}
 #endif //end of CONFIG_ZU67_SELECTION
 
-	printf("%s: base_addr %#x *** fpga_data_addr : %#x - %#x \n", 
+	printf("%s: base_addr %#x *** fpga_data_addr : %#x - %#x \n",
 		__func__, base_addr, fpga_data_addr, fpga_data_size);
 	/* Reserved 10M, */
 	unzip_data_addr = fpga_data_addr + fpga_data_size + SZ_60M;
@@ -592,6 +606,7 @@ static int do_zboot(struct cmd_tbl *cmdtp, int flag, int argc, char * const argv
 	clu_init();
 #endif
 
+kernel_loading:
 	image_addr = base_addr + sh->head_size + sh->kernel_offset;
 
 	/* boot kernel */
