diff --git a/include/configs/xilinx_versal.h b/include/configs/xilinx_versal.h
index 64f123424cf..45c799c3c79 100644
--- a/include/configs/xilinx_versal.h
+++ b/include/configs/xilinx_versal.h
@@ -10,6 +10,8 @@
 #ifndef __XILINX_VERSAL_H
 #define __XILINX_VERSAL_H
 
+#include <zboot_info.h>
+
 /* Generic Interrupt Controller Definitions */
 #define GICD_BASE	0xF9000000
 #define GICR_BASE	0xF9080000
@@ -30,6 +32,8 @@
 /* Ethernet driver */
 
 #define ENV_MEM_LAYOUT_SETTINGS \
+	"initrd_high=0x28000000\0" \
+	"zboot=1\0" \
 	"fdt_addr_r=0x40000000\0" \
 	"fdt_size_r=0x400000\0" \
 	"pxefile_addr_r=0x10000000\0" \
@@ -37,8 +41,14 @@
 	"kernel_size_r=0x10000000\0" \
 	"kernel_comp_addr_r=0x30000000\0" \
 	"kernel_comp_size=0x3C00000\0" \
+	"scriptaddr=0x2000000\0" \
 	"ramdisk_addr_r=0x02100000\0" \
 	"script_size_f=0x80000\0"
+	"script_offset_f=0x7e0000\0" \
+	"script_size_f=0x20000\0" \
+	"zboot_info=" ZBOOT_INFO "\0" \
+	"zboot_ver=" ZBOOT_VERSION "\0" \
+	"bypass_fpga_load=yes\0"
 
 #if defined(CONFIG_DISTRO_DEFAULTS)
 
