diff --git a/cmd/Kconfig b/cmd/Kconfig
index 69fad7f817..59e95e9cba 100644
--- a/cmd/Kconfig
+++ b/cmd/Kconfig
@@ -966,6 +966,12 @@ config ZU67_SELECTION
 	help
 	  Automatically loads the zu67 limited and standard bit files.
 
+config ZU67_SE_FPGA_LOAD
+	bool "zu67-se fpga load"
+	depends on CMD_ZBOOT_Z
+	help
+	  zu67-se does not use encrypted bit files.
+
 menu "Device access commands"
 
 config CMD_ARMFLASH
diff --git a/cmd/zboot.c b/cmd/zboot.c
index f31dbe3fbb..70b8bd1517 100644
--- a/cmd/zboot.c
+++ b/cmd/zboot.c
@@ -220,7 +220,7 @@ static int do_fpga_loadb(uint32_t fpgadata, uint32_t datasize)
 		return -1;
 	}
 
-#ifdef CONFIG_ZBOOT_RSA_BITSTREAM
+#if defined(CONFIG_ZBOOT_RSA_BITSTREAM) && !defined(CONFIG_ZU67_SE_FPGA_LOAD)
 	return do_fpga_loads_rsa_bitstream(dev, fpgadata, datasize);
 #else
 	return fpga_loadbitstream(dev, (void *)fpgadata, datasize, BIT_FULL);
