diff --git a/cmd/zboot.c b/cmd/zboot.c
index dc7a4cf367..805bebdc7e 100644
--- a/cmd/zboot.c
+++ b/cmd/zboot.c
@@ -174,6 +174,15 @@ static long get_fpga_device(char *arg)
 	return dev;
 }
 
+static int process_bypass_fpga_load(void)
+{
+    const char *val = env_get("bypass_fpga_load");
+    if (val && strcmp(val, "yes") == 0) {
+        return 1;
+    } 
+	return 0;
+}
+
 static int do_fpga_loads_rsa_bitstream(int devnum, uint32_t fpga_data, uint32_t data_size)
 {
 	struct fpga_secure_info fpga_sec_info;
@@ -447,6 +456,11 @@ static int do_zboot(struct cmd_tbl *cmdtp, int flag, int argc, char * const argv
 	LEGAL_ADDR(base_addr, dataptr);
 
 	sh = (slot_head *)dataptr;
+	
+	if(process_bypass_fpga_load()) {
+		printf("bypass_fpga_load is set to 'yes', proceeding with FPGA loading bypass.\n");
+		goto kernel_loading;
+	}
 
 #ifdef CONFIG_ZU67_SELECTION
 	int zu67_selection = 1;//set standard zu67 as default.
@@ -591,6 +605,7 @@ static int do_zboot(struct cmd_tbl *cmdtp, int flag, int argc, char * const argv
 	clu_init();
 #endif
 
+kernel_loading:
 	image_addr = base_addr + sh->head_size + sh->kernel_offset;
 
 	/* boot kernel */
