diff --git a/cmd/ethloop.c b/cmd/ethloop.c
index 35dee1f2144..e48fe27f6c0 100644
--- a/cmd/ethloop.c
+++ b/cmd/ethloop.c
@@ -15,6 +15,7 @@ static int do_loopback_assert(int argc, char *const argv[])
 	loopback_state_set(1);
 	net_loop(LOOPBACK);
 	loopback_state_set(0);
+	net_loop(LOOPBACKEXIT);
 
 	return CMD_RET_SUCCESS;
 }
diff --git a/drivers/net/phy/motorcomm.c b/drivers/net/phy/motorcomm.c
index 264b51a3fcb..b5cbc59b677 100644
--- a/drivers/net/phy/motorcomm.c
+++ b/drivers/net/phy/motorcomm.c
@@ -356,14 +356,21 @@ int yt8521S_loopback_exit(struct phy_device *phydev)
 {
 	int ret = 0;
 	printf("yt8521S_loopback_exit\n");
+#if 0
 	ret = yt_write_ext(phydev, 0xa0c0, 0x8000);
 	if (ret < 0){
 		printf("yt_write_ext 0xa0c0 fail:%d.\n", ret);
 		return ret;
 	}
-
-	if (phy_read(phydev, MDIO_DEVAD_NONE, MII_BMCR) == 0x1000)
-		return 0;
+#endif
+	int val;
+	val = yt_read_ext(phydev, 0x0A);
+	val &= ~(1<<4);
+	ret = yt_write_ext(phydev, 0x0A, val);
+	if (ret < 0){
+		printf("yt_write_ext 0x0A fail:%d.\n", ret);
+		return ret;
+	}
 
 	ret = phy_write(phydev, MDIO_DEVAD_NONE, MII_BMCR, 0x9000);
 	if (ret < 0) {
diff --git a/include/net-legacy.h b/include/net-legacy.h
index 0885e778004..74c0c520b7f 100644
--- a/include/net-legacy.h
+++ b/include/net-legacy.h
@@ -305,7 +305,7 @@ extern int		net_restart_wrap;	/* Tried all network devices */
 enum proto_t {
 	BOOTP, RARP, ARP, TFTPGET, DHCP, DHCP6, PING, PING6, DNS, NFS, CDP,
 	NETCONS, SNTP, TFTPSRV, TFTPPUT, LINKLOCAL, FASTBOOT_UDP, FASTBOOT_TCP,
-	WOL, UDP, NCSI, WGET, RS, LOOPBACK
+	WOL, UDP, NCSI, WGET, RS, LOOPBACK, LOOPBACKEXIT
 };
 
 /* Indicates whether the file name was specified on the command line */
diff --git a/net/net.c b/net/net.c
index dd3b00a6140..73ba9da0813 100644
--- a/net/net.c
+++ b/net/net.c
@@ -463,11 +463,17 @@ int net_loop(enum proto_t protocol)
 		eth_set_current();
 		ret = eth_init();
 		if (ret < 0) {
+#ifdef CONFIG_LOOP_BACK_ASSERT
+			if (protocol == LOOPBACKEXIT) {
+				printf("%s: act loopback mode exit!\n", __func__);
+				return ret;
+			}
 			if (!env_get("Loopback") || strcmp(env_get("Loopback"), "no") != 0) {
 				env_set("Loopback", "no");
 				env_set("bootenv_save", "true"); //save env later;
 			}
 			printf("Loopback NOT detected!\n");
+#endif
 			eth_halt();
 			return ret;
 		}
@@ -535,6 +541,9 @@ restart:
 		case LOOPBACK:
 			loopback_start_server();
 			break;
+		case LOOPBACKEXIT:
+			printf("%s: do loopback mode exit!\n", __func__);
+			break;
 #endif
 #if defined(CONFIG_CMD_DHCP)
 		case DHCP:
@@ -1497,6 +1506,11 @@ static int net_check_prereq(enum proto_t protocol)
 		}
 		goto common;
 #endif
+#if defined(CONFIG_LOOP_BACK_ASSERT)
+	case LOOPBACKEXIT:
+		printf("%s: loopback mode exit!\n", __func__);
+		return 1;
+#endif
 #if defined(CONFIG_CMD_PING6)
 	case PING6:
 		if (ip6_is_unspecified_addr(&net_ping_ip6)) {
