diff --git a/include/configs/xilinx_zynqmp.h b/include/configs/xilinx_zynqmp.h
index c8ce7055265..c87c03ea955 100644
--- a/include/configs/xilinx_zynqmp.h
+++ b/include/configs/xilinx_zynqmp.h
@@ -47,7 +47,7 @@
 /* Ethernet driver */
 
 #define ENV_MEM_LAYOUT_SETTINGS \
-	"initrd_high=0x38000000\0" \
+	"initrd_high=0x28000000\0" \
 	"zboot=1\0" \
 	"fdt_addr_r=0x40000000\0" \
 	"fdt_size_r=0x400000\0" \
