diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_config.h b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_config.h
index d6d8dc231f..ec2088e8c1 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_config.h
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_config.h
@@ -77,7 +77,7 @@ extern "C" {
 #endif
 
 #ifndef FSBL_DEBUG_VAL
-#define FSBL_DEBUG_VAL              (0U)
+#define FSBL_DEBUG_VAL              (1U)
 #endif
 
 #ifndef FSBL_DEBUG_INFO_VAL
@@ -152,7 +152,7 @@ extern "C" {
 #endif
 
 #ifndef FSBL_SECURE_EXCLUDE_VAL
-#define FSBL_SECURE_EXCLUDE_VAL			(0U)
+#define FSBL_SECURE_EXCLUDE_VAL			(1U)
 #endif
 
 #ifndef FSBL_BS_EXCLUDE_VAL
@@ -192,7 +192,7 @@ extern "C" {
 #endif
 
 #ifndef FSBL_FORCE_ENC_EXCLUDE_VAL
-#define FSBL_FORCE_ENC_EXCLUDE_VAL		(0U)
+#define FSBL_FORCE_ENC_EXCLUDE_VAL		(1U)
 #endif
 
 #ifndef FSBL_DDR_SR_EXCLUDE_VAL
diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_initialization.c b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_initialization.c
index 39a219efe0..406f19cc2c 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_initialization.c
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_initialization.c
@@ -1244,9 +1244,7 @@ static u32 XFsbl_ValidateHeader(XFsblPs * FsblInstancePtr)
 	}
 
 	/* If authentication is enabled */
-	if (((EfuseCtrl & EFUSE_SEC_CTRL_RSA_EN_MASK) != 0U) ||
-	    ((BootHdrAttrb & XIH_BH_IMAGE_ATTRB_RSA_MASK)
-		== XIH_BH_IMAGE_ATTRB_RSA_MASK)) {
+	if (0) {
 		FsblInstancePtr->AuthEnabled = TRUE;
 		XFsbl_Printf(DEBUG_INFO,"Authentication Enabled\r\n");
 #ifdef XFSBL_SECURE
@@ -1258,6 +1256,9 @@ static u32 XFsbl_ValidateHeader(XFsblPs * FsblInstancePtr)
 			XFsbl_Printf(DEBUG_GENERAL,"Device Copy Failed \n\r");
 			goto END;
 		}
+
+		XFsbl_Printf(DEBUG_INFO,"AcOffset:%d\r\n", AcOffset);
+
 		if (AcOffset != 0x00U) {
 			/* Authentication exists copy AC to OCM */
 			Status = FsblInstancePtr->DeviceOps.DeviceCopy(
@@ -1267,10 +1268,13 @@ static u32 XFsbl_ValidateHeader(XFsblPs * FsblInstancePtr)
 			if (XFSBL_SUCCESS != Status) {
 				goto END;
 			}
+			
+			XFsbl_Printf(DEBUG_INFO,"EfuseCtrl:0x%04x\r\n", EfuseCtrl);
 
 			/* Authenticate boot header */
 			/* When eFUSE RSA enable bit is blown */
 			if ((EfuseCtrl & EFUSE_SEC_CTRL_RSA_EN_MASK) != 0U) {
+				XFsbl_Printf(DEBUG_INFO," eFUSE RSA enable bit is blown\r\n");
 				Status = XFsbl_BhAuthentication(FsblInstancePtr,
 						ReadBuffer, (PTRSIZE)AuthBuffer, TRUE);
 			}
@@ -1285,6 +1289,7 @@ static u32 XFsbl_ValidateHeader(XFsblPs * FsblInstancePtr)
 				goto END;
 			}
 
+			XFsbl_Printf(DEBUG_INFO,"Authenticate Image header table\r\n");
 
 			/* Authenticate Image header table */
 			/*
@@ -1299,6 +1304,7 @@ static u32 XFsbl_ValidateHeader(XFsblPs * FsblInstancePtr)
 				Status = XFSBL_ERROR_IMAGE_HEADER_SIZE;
 				goto END;
 			}
+			XFsbl_Printf(DEBUG_INFO,"Copy the Image header to OCM\r\n");
 
 			/* Copy the Image header to OCM */
 			Status = FsblInstancePtr->DeviceOps.DeviceCopy(
@@ -1308,6 +1314,7 @@ static u32 XFsbl_ValidateHeader(XFsblPs * FsblInstancePtr)
 			if (Status != XFSBL_SUCCESS) {
 				goto END;
 			}
+			XFsbl_Printf(DEBUG_INFO,"Authenticate the image header\r\n");
 
 			/* Authenticate the image header */
 			Status = XFsbl_Authentication(FsblInstancePtr,
