diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_initialization.c b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_initialization.c
index 39a219efe0..745e172801 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_initialization.c
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_initialization.c
@@ -1580,7 +1580,7 @@ static u32 XFsbl_SecondaryBootDeviceInit(XFsblPs * FsblInstancePtr)
 		break;
 
 	}
-
+	goto END;
 	/**
 	 * Initialize the Secondary Boot Device Driver
 	 */
diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c
index d38f799e91..7166bb58fc 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c
@@ -289,7 +289,7 @@ int main(void )
 
 		case XFSBL_STAGE_ERR:
 			{
-				XFsbl_Printf(DEBUG_INFO,
+				XFsbl_Printf(DEBUG_PRINT_ALWAYS,
 						"================= In Stage Err ============ \n\r");
 
 				XFsbl_ErrorLockDown(FsblStatus);
diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_misc_drivers.c b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_misc_drivers.c
index fba2d7ca4f..4b2f38c732 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_misc_drivers.c
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_misc_drivers.c
@@ -336,8 +336,8 @@ u32 XFsbl_PmInit(void)
 	}
 #endif /** end of IPI related code */
 #endif /* End of XPAR_XILPM_ENABLED */
-	XFsbl_Printf(DEBUG_DETAILED,"PM Init Success\r\n");
-	UStatus = XFSBL_SUCCESS;
+	XFsbl_Printf(DEBUG_PRINT_ALWAYS,"PM Init Failed\r\n");
+	UStatus = XFSBL_ERROR_PM_INIT;
 END:
 	return UStatus;
 }
