diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_initialization.c b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_initialization.c
index 39a219efe0..e794ce9286 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_initialization.c
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_initialization.c
@@ -936,8 +936,8 @@ static u32 XFsbl_PrimaryBootDeviceInit(XFsblPs * FsblInstancePtr)
 		break;
 		case XFSBL_JTAG_BOOT_MODE:
 		{
-			XFsbl_Printf(DEBUG_GENERAL,"In JTAG Boot Mode \n\r");
-			Status = XFSBL_STATUS_JTAG;
+			XFsbl_Printf(DEBUG_GENERAL,"Enter JTAG Boot Mode Failed!\n\r");
+			Status = XFSBL_ERROR_UNSUPPORTED_BOOT_MODE;
 		}
 		break;
 
